docs/data/sdks/next-gen-sdks.md

```diff
@@ -0,0 +1,186 @@
+
+You can pass a Configuration when Client initialization.
+
+```python
+# Next-gen Python SDK example: initialize a Client with a Configuration
+
+from amplitude import Amplitude
+from amplitude import Config
+
+amplitude = Amplitude("API_KEY", 
+                      configuration=Config(
+                            flush_queue_size=100, # flush queued events when there are 100 or more
+                            flush_interval_millis=20000 # flush queued events every 20 seconds
+                    )) 
+```
+
+You can also get access to the Configuration of your Client after initialization.
+
+```python
+Next-gen Python SDK example: get access to the Configuration of your Client
+
+from amplitude import Amplitude
+
+client = Amplitude('API_KEY')
+# flush queued events when there are 100 or more
+client.configuration.flush_queue_size = 100
+# flush queued events every 20 seconds
+client.configuration.flush_interval_millis = 20000
+```
+
+### Storage provider
+
+A storage provider holds events in a designated storage buffer. The default storage provider only queues events in memory. To prevent data lose, you can customize your own storage provider to persist events, for example to a disk, in case of unexpected shutdown. This way, you can ensure that you can recover un-sent events even in the event of a system failure. 
+
+You can configure the storage provider in the Configuration of your Client. Refer to the "Configuration Options" section of a specific next-gen SDK full documentation for more details.
+
+### Logger provider
+
+A logger provider configure the logger instance used by Amplitude Client. It helps to collect debugging and error messages from the SDK in both development and production environments.
+
+You can configure the logger provider in the Configuration of your Client. Refer to the "Configuration Options" section of a specific next-gen SDK full documentation for more details.
+
+### Event 
+
+Events represent how users interact with your application. 
+
+`BaseEvent` represents a basic event with optional properties. You can also track other event properties in the field of `EventProperties`
+
+```python
+# Next-gen Python SDK example: Event
+
+from amplitude import Amplitude, BaseEvent
+
+client = Amplitude('API_KEY')
+
+# Track a basic event
+# One of user_id and device_id is required
+event = BaseEvent(event_type="Button Clicked", user_id="USER_ID")
+client.track(event)
+
+# Track events with optional properties
+client.track(
+    BaseEvent(
+        event_type="Button Clicked",
+        user_id="USER_ID",
+        app_version="1.2.1" # optional, the current version of your application
+        event_properties={ # other event properties you want to track 
+            "color": "red"
+        }
+))
+```
+
+### Plugin
+
+Plugins allow you to extend Amplitude SDK's behavior by, for example:
+
+- Use an Enrichment Plugin to modify event properties
+- Use a Destination Plugin to send events to a third-party APIs
+
```

Comment:
We should probably mention the order of plugins again in this section: Enrichment -> Destinations. And that all enrichment plugins run in the order they were added, then all destinations in order then were added.

Reply:
Added a explanation of execution order